### ISSUE TITLE
Feature request: Allow for recurring events import

### ISSUE THREAD

**Issue by lsdjr72** [haydenull/logseq-plugin-agenda#191]
Until this plug-in can allow recurring events to be imported from external calendars, it's not going to be useful for me. I can't spend my time comparing two calendars (agenda and Google). But if all repeating events could be imported into Agenda, then I would use it exclusively because of the ability to schedule tasks from projects.

**Comment by mcopik** [comment]
Seconding this issue. Unfortunately, the lack of repeated events in calendar subscriptions is a severe limitation.

How difficult would be adding this feature? I would be happy to help contribute to this feature. [Looking at the current code](https://github.com/haydenull/logseq-plugin-agenda/blob/2977810df16da956afc7155ad724b98dce82294f/src/util/subscription.ts#L74), agenda is not using `ical` model layer but using the package only to parse calendar's contents. However, `ical` can generate all occurrences from `rrule`, as long as we use the package's component layer.

For example, this code using `ical.js` generates the first 5 occurrences of the scheduled event. Are there other major issues that might prevent us from using recurring events?

```node
let component = new ical.Component(ical.parse(calendar_data));
const vevents = component.getAllSubcomponents('vevent').map((c) => {

  vevent = new ical.Event(c);
  const iterator = vevent.iterator();
  let next;
  let num = 0;
  while ((next = iterator.next()) && num++ < 5) {
     console.log(String(vevent.summary))
     console.log(vevent.getOccurrenceDetails(next).startDate.toICALString());
  }

});
```

**Comment by haydenull** [comment]
I mainly focus on how to record recurring events in logseq, and how to make it compatible with todoist and the build-in solutions of logseq.

The code you mentioned here is for supporting recurring events in calendar subscription. There shouldn't be any problem with it, just need to test the compatibility on several mainstream platforms.

**Comment by mcopik** [comment]
@haydenull Yes, I interpreted this issue as related to importing recurring events from the external calendar: "recurring events to be imported from external calendars".

I don't have any means of testing with iCloud calendar, but I'm happy to test it against Google Calendar and start a PR later.

**Comment by haydenull** [comment]
@mcopik That's great! I can test iCloud calendar on my end.

**Comment by AnweshGangula** [comment]
Currently I can see that recurring events are working fine from google calendar, but not from outlook
